cmake/optional.cmake

```diff
@@ -57,3 +57,29 @@ if(WITH_MINIUPNPC)
     message(FATAL_ERROR "libminiupnpc requested, but not found.")
   endif()
 endif()
+
+if(WITH_ZMQ)
```

Comment:
zmq started shipping a cmake config file starting with v4.2.2.

Once we bump our zmq requirement in the future, we'll be able to use `find_package(ZeroMQ)`. Mind adding a note?

Reply:
> zmq started shipping a cmake config file starting with v4.2.2.

But not Ubuntu's libzmq3-dev packages, unfortunately.

Reply:
Still, mind adding a note that it's supported upstream and we'll be able to switch at some point in the future?

Reply:
> Still, mind adding a note that it's supported upstream and we'll be able to switch at some point in the future?

Sure. I'll add it.

Reply:
[Updated](https://github.com/hebasto/bitcoin/pull/13#issuecomment-1507244320).